cookbook/basic_rag_example/pinecone_embeddings.py

```diff
@@ -0,0 +1,125 @@
+"""A script using RAG and Pinecone to summarize relevant articles from a large database.
+
+First we load in a dataset of news articles from 2004 and embed them. Then we use
+Pinecone to efficiently find the articles with most relevance to a query, then use said
+articles in a prompt to get the summary.
+"""
+import ast
+import os
+
+import pandas as pd
+import tiktoken
+from pinecone import Pinecone, ServerlessSpec
+
+from mirascope import OpenAIChat, Prompt
+
+os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+chat = OpenAIChat()
+embeddings_model = "text-embedding-ada-002"
+
+if not os.path.exists("cookbook/basic_rag_example/news_article_dataset.csv"):
+    url = "https://raw.githubusercontent.com/Dawit-1621/BBC-News-Classification/main/Data/BBC%20News%20Test.csv"
+    df = pd.read_csv(url)
+    split_articles = []
+    encoder = tiktoken.encoding_for_model("gpt-3.5-turbo")
+
+    for i, row in df.iterrows():
```

Comment:
make this into a helper function with a docstring that explains it

Reply:
this should be something like `df = new_function(df)`

Covers lines 26-40

Also make sure to add a comment explaining lines 40-42 in the new function